### PR TITLE
#50: Ignore unkown options in down()

### PR DESCRIPTION
**Pull request by jukkah**
Fixed problem found in 7cc0f2f.

### CODE REVIEW

**Comment by tremby** [comment]
It seems rather overcomplicated. What about this?

```js
// Deal with special case where an array has been passed
if (Array.isArray(options)) {
  return this._run('down', options, getExecuted.bind(this));
}

if (typeof options === "undefined" || options === null || options.to == null) {
  return getExecuted().bind(this).then(function(migrations) {
    if (migrations[0]) {
      return this.down(migrations[0].file);
    }
    return Bluebird.resolve([]);
  });
}
```

Untested.

**Comment by tremby** [comment]
No, ignore the above. It doesn't handle some things properly. It think this would be better handled in the _run method.

**Comment by tremby** [comment]
I was thinking something more along the lines of this: https://github.com/tremby/umzug/commit/0db16eba7a754542504b570d4a226d306f0fcc58

It's untested (see #65); let me know what you think. It may well be incomplete.

**Comment by tremby** [comment]
You can ignore that patch for now; it's entirely broken. Too tired to hack on it right now, though.

**Comment by jukkah** [comment]
Even if it is broken, I got your point :sunglasses:

**Comment by jukkah** [comment]
@sdepold Your review is needed.

File: test/index/down.test.js [hunk 1]
```diff
@@ -77,27 +77,6 @@ describe('down', function () {
 
     describe('when no option is specified', function () {
       beforeEach(function () {
-        return this.umzug.down().bind(this).then(function (migrations) {
-          this.migrations = migrations;
-        });
-      });
-
-      it('returns 1 item', function () {
```
Comment (sdepold) [inline test/index/down.test.js]: why are we deleting tests?

Comment (jukkah) [inline test/index/down.test.js]: I have considered `down()` and `down({})` are the same test cases. In addition I'm refactoring this completely.

**Comment by jukkah** [comment]
I give up with this as I'm working on v2.0 now.

**Pull request closed without merging**